spf.go

```diff
@@ -65,32 +66,51 @@ func newMissingMacrosError(domain string, macros []string) error {
 // could be totally ignored by implementation.
 type IPMatcherFunc func(ip net.IP, name string) (bool, error)
 
-// Resolver provides abstraction for DNS layer
+// ResponseExtras contains additional information returned alongside DNS query results.
+type ResponseExtras struct {
+	TTL  time.Duration // Minimum TTL of the DNS response
+	Void bool          // Indicates if the response is a result of a DNS void lookup.
+
+	// A DNS void lookup, as defined in Section 4.6.4 of RFC 7208 (https://datatracker.ietf.org/doc/html/rfc7208#section-4.6.4),
+	// is a query for a domain that is intentionally configured to have no associated DNS records,
+	// such as an explicit configuration for a "blackhole" or an intentionally nonexistent domain.
+	// This type of query typically returns a response with no relevant DNS records (e.g., NXDOMAIN),
+	// and the 'Void' field in this struct is set to 'true' to indicate that the response resulted from such a lookup.
+}
```

Comment:
You pass ResponseExtras as a reference everywhere to use a nil value when some non-DNS error occurs. This is ok, but it forces users to do additional checks for nil. It increases pollution. Making fields private and using methods that will consider nil receivers would be beneficial. Like
```golang
func (x *ResponseExtras) TTL() time.Duration {
  if x==nil { return 0 }
  return x.ttl
}

func (x *ResponseExtras) IsVoid() bool {
   return x!=nil && x.void
}
```

Reply:
done